src/store/index-level.ts

```diff
@@ -278,29 +528,20 @@ export class IndexLevel {
   }
 
   /**
-   * Extracts the value encoded within the indexed key when a record is inserted.
+   * Encodes an indexed value to a string
    *
-   * ex. key: 'dateCreated\u0000"2023-05-25T18:23:29.425008Z"\u0000bafyreigs3em7lrclhntzhgvkrf75j2muk6e7ypq3lrw3ffgcpyazyw6pry'
-   *     extracted value: "2023-05-25T18:23:29.425008Z"
-   *
-   * @param key an IndexLevel db key.
-   * @returns the extracted encodedValue from the key.
-   */
-  static extractValueFromKey(key: string): string {
-    const [, value] = key.split(this.delimiter);
-    return value;
-  }
-
-  /**
-   * Joins the given values using the `\x00` (\u0000) character.
+   * NOTE: we currently only use this for strings, numbers and booleans.
    */
-  private static delimiter = `\x00`;
-  private join(...values: unknown[]): string {
-    return values.join(IndexLevel.delimiter);
+  static encodeValue(value: string | number | boolean): string {
+    switch (typeof value) {
+    case 'string':
+      // We can't just `JSON.stringify` as that'll affect the sort order of strings.
+      // For example, `'\x00'` becomes `'\\u0000'`.
```

Comment:
Should we remove? This comment doesn't make sense to me as we don't deal with \x00 here. Maybe lost meaning due to iterations of refactoring?